src/components/Maps/MapPage.tsx

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import { useQuery } from 'react-apollo';
+import GoogleMapReact from 'google-map-react';
+
+import MenuAppBar from '../common/MenuAppBar';
+import InfoWindow from './InfoWindow';
+import { LocationPin, GET_PINS_BY_EVENT_ID } from '../../graphql/queries/maps';
+
+const MapPage = ({
+  match: {
+    params: { eventId },
+  },
+}: {
+  match: { params: { eventId: string } };
+}) => {
+  const defaultMap = {
+    center: { lat: 43.470846, lng: -80.538473 },
+    zoom: 11,
+  };
+
+  const getMapOptions = (maps) => {
+    return {
+      streetViewControl: false,
+      scaleControl: true,
+      fullscreenControl: false,
+      styles: [
+        {
+          featureType: 'poi.business',
+          elementType: 'labels',
+          stylers: [
+            {
+              visibility: 'off',
+            },
+          ],
+        },
+      ],
+      gestureHandling: 'greedy',
+      disableDoubleClickZoom: true,
+
+      mapTypeControl: true,
+      mapTypeId: maps.MapTypeId.ROADMAP,
+      mapTypeControlOptions: {
+        style: maps.MapTypeControlStyle.HORIZONTAL_BAR,
+        position: maps.ControlPosition.BOTTOM_CENTER,
+        mapTypeIds: [
+          maps.MapTypeId.ROADMAP,
+          maps.MapTypeId.SATELLITE,
+          maps.MapTypeId.HYBRID,
+        ],
+      },
+
+      zoomControl: true,
+      clickableIcons: false,
+    };
+  };
+
+  const { data, loading } = useQuery(GET_PINS_BY_EVENT_ID, {
+    variables: { eventId },
+  });
+
+  const pins: Array<LocationPin> = data && !loading ? data.pinsForEvent : [];
+  const [infoWindowOpen, setInfoWindowOpen] = React.useState(false);
+  const [interestPinTitle, setInterestPinTitle] = React.useState('');
+  const [interestPinLocation, setInterestPinLocation] = React.useState('');
+
+  const initMaps = (map, maps) => {
+    if (!loading) {
```

Comment:
I occasionally run into an issue where the pins don't load when I refresh the page. If you are still running into this issue, let me know. I have tried many fixes and I believe the error is somewhere in this function. Hopefully I have addressed it.